apps/gatsby/src/Sidebar.js

```diff
@@ -42,8 +40,72 @@ export default class Sidebar extends React.Component {
     this.debounceInterval = setInterval(this.refreshPreview, 1000);
   };
 
-  componentDidMount() {
+  buildSlug = async () => {
+    const {urlConstructors, previewUrl} = this.sdk.parameters.installation;
+    //Find the url constructor for the given contentType
+    const constructor = urlConstructors ? urlConstructors.find(
+      constructor => constructor.id === this.sdk.contentType.sys.id
+    ) : undefined;
+    // If there is no constructor set the url as the base preview
+    if (!constructor){
+      return
+    }
+
+    //QUESTION: What does the locale object in the sdk look like when there are multiple locales available. Need to account for spaces that have more than one locale, so they can have previews everywhere.
+
+    // Recursive helper to return slug values buried in a chain of references
+    const resolveReferenceChain = async (sdk, array, index, parentId) => {
+      console.log(sdk)
+      // Full entry to access child fields
+      const fullParentEntry = await sdk.space.getEntry(parentId)
+      // Child field
+      const childField = fullParentEntry.fields[array[index + 1]][sdk.locales.default]
+      if (Array.isArray(childField)) {
+       console.error("Gatsby Preview App: You are trying to search for a slug in a multi reference field. Only single reference fields are searchable with this app. Either change the field to a single reference, or change the field you are searching for in the slug.")
+        return ""
+      }
+
+      if (index + 2 < array.length) {
+        return resolveReferenceChain(sdk, array, (index + 1), childField.sys.id)
+      } else {
+        return childField
+      }
+    }
```

Comment:
Do we need to redefine this on every mount? 🤔 Maybe it can be just a method of the class